### DISCUSSION TITLE
Add access control to each repository's branches

### DISCUSSION THREAD

**Discussion by ghost** [community/community#16334] [Other Feature Feedback, Questions, & Ideas]
Hi all! How are you guys?

#### 1. Feature name
Control access to the branches of each repository

#### 2. Feature function
1. What problem are you trying to solve by requesting this feature?
    - So... I'd like to be able to manage or have private branches in a public repository, as well as public branches in private repositories. I would like to control access to the branches of each repository. But... this is currently not possible.  So... as far as I know, at the moment there are only private repositories and public repositories in the free plans, this free plan is what I am currently on. 
2. What will this feature do differently?
     - This will allow we can have an - private repository with a public branch and public repository with a private branch.
3. What benefits will this feature bring?
   - This allows developers or companies to have open source and closed source together in the same repository without necessarily having to make separate private repositories for this purpose.
4. Do you have any idea where this will be affected?
   -  Web
5. What is your argument for requesting this feature?
    - If this feature is accepted, part of the open source projects will have a better management position. Because... this would allow multiple GitHub projects to adopt the [Open-core model ](https://en.wikipedia.org/wiki/Open-core_model) - part of the core is open, another part is closed. This would be done with the access control of the branches of each repository.
    - Also, it is very complicated to manage multiple separate repositories when we can manage access control for each branch for each specific repository. I believe it to be an interesting, efficient option. 

#### 3. ui/ux
![image](https://user-images.githubusercontent.com/95204402/168206906-edd7c1b3-d22c-4776-a4d7-852e9be7cf71.png)

#### 4. Image description
With the branch control, you can search by branch types: public, private

#### 5. Notes
1. I don't know if there is such a feature for the paid GitHub service, I don't have that knowledge right now. So... If anyone has any feedback on this, I would be glad for the answer, feedback. I would like positive or negative feedback on this idea, this feature that I request, I would be happy if there was an answer about this, about this idea, this feature.
2. I believe the best platform in the world is GitHub. I wish I had this feature... because this could help me, it could help the GitHub community and businesses.
3. I don't know if this feature would be possible... I don't know if Git allows it... anyway, I'm happy with feedback if anyone reads and comments here the criticism or improvement suggestion.
4. I don't know if there is any company that has this feature, if it's accepted by GitHub and the GitHub community would be happy. It can be an innovative feature - make GitHub more competitive on the market.
5. I'm not criticizing the management of GitHub just thinking about how to make GitHub better - if I can... because GitHub is just too wonderful.
6. There are several people requesting the same resource or else I think they are very similar resources - or resources they have certain common characteristics - I can be right or wrong, either way here is the reference: [Search in branches](https://github.com/github/feedback/discussions/16972) , [Add branch as a search filter](https://github.com/github/feedback/discussions/8564)

**Comment by Neustradamus** [comment]
At this time, in 2025, I have always the problem, the search is only with default branch.

It is possible to add a solution?

Thanks in advance.
